doit/cmd_base.py

```diff
@@ -310,30 +310,64 @@ def _load_from(cmd, namespace, cmd_list):
         return task_list, doit_config
 
 
-class ModuleTaskLoader(TaskLoader):
+class TaskLoader2(TaskLoader):
+    """Interface of task loaders with new-style API.
+
+    The default implementation assumes tasks are loaded from a namespace, mapping identifiers to
+    elements like functions (like task generators) or constants (like configuration values).
+
+    This API update separates the loading of the configuration and the loading of the actual tasks,
+    which enables additional elements to be available during task creation.
+    """
+    def __init__(self):
+        super().__init__()
+        self.namespace = None
+
+    def setup(self, opt_values):
+        """Delayed initialization.
+
+        To be implemented if the data is needed by derived classes.
+
+        :param opt_values: (dict) with values for cmd_options
+        """
+
+    def load_doit_config(self):
+        """Load doit configuration."""
+        return loader.load_doit_config(self.namespace)
+
+    def load_tasks(self, cmd, pos_args):
+        """Load tasks.
+
+        :return: (List[Task])
```

Comment:
Not sure it is a standard but usually I put ``return`` description *after*  ``param``